pkg/samples/fixtures.go

```diff
@@ -0,0 +1,283 @@
+package samples
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/joho/godotenv"
+	"github.com/spf13/afero"
+	"github.com/thedevsaddam/gojsonq"
+
+	"github.com/stripe/stripe-cli/pkg/requests"
+)
+
+// SupportedVersions is the version number of the fixture template the CLI supports
+const SupportedVersions = 0
+
+type metaFixture struct {
+	Version int `json:"_version"`
+}
+
+type fixtureFile struct {
+	Meta     metaFixture       `json:"_meta"`
+	Fixtures []fixture         `json:"fixtures"`
+	Env      map[string]string `json:"env"`
+}
+
+type fixture struct {
+	Name   string      `json:"name"`
+	Path   string      `json:"path"`
+	Method string      `json:"method"`
+	Params interface{} `json:"params"`
+}
+
+// Fixture contains a mapping of an individual fixtures responses for querying
+type Fixture struct {
+	Fs        afero.Fs
+	APIKey    string
+	BaseURL   string
+	responses map[string]*gojsonq.JSONQ
+	fixture   fixtureFile
+}
+
+// NewFixture creates and executes a fixtures steps for populating test data
+func (fxt *Fixture) NewFixture(file string) error {
+	fxt.responses = make(map[string]*gojsonq.JSONQ)
+
+	filedata, err := afero.ReadFile(fxt.Fs, file)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(filedata, &fxt.fixture)
+	if err != nil {
+		return err
+	}
+
+	if fxt.fixture.Meta.Version > SupportedVersions {
+		return fmt.Errorf("Fixture version not supported: %s", string(fxt.fixture.Meta.Version))
+	}
+
+	return nil
+}
+
+// Execute takes the parsed fixture file and runs through all the requests
+// defined to populate the user's account
+func (fxt *Fixture) Execute() error {
+	for _, data := range fxt.fixture.Fixtures {
+		fmt.Println(fmt.Sprintf("Setting up fixture for: %s", data.Name))
+
+		resp, err := fxt.makeRequest(data)
+		if err != nil {
+			return err
+		}
+
+		fxt.responses[data.Name] = gojsonq.New().FromString(string(resp))
+	}
+
+	return nil
+}
+
+// UpdateEnv uses the results of the fixtures command just executed and
+// updates a local .env with the resulting data
+func (fxt *Fixture) UpdateEnv() error {
+	if len(fxt.fixture.Env) > 0 {
+		return fxt.updateEnv(fxt.fixture.Env)
+	}
+
+	return nil
+}
+
+func (fxt *Fixture) makeRequest(data fixture) ([]byte, error) {
+	var rp requests.RequestParameters
+	if data.Method == "post" {
+		now := time.Now().String()
+		metadata := fmt.Sprintf("metadata[_created_by_fixture]=%s", now)
```

Comment:
Some resources might not accept metadata, so this is slightly risky. We can keep this as a documented limitation for now, but long term we should check the OpenAPI spec to decide whether or not to include metadata.

Reply:
I thought about that too, would the API fail here if we passed metadata to a resource that doesn't support it?

Reply:
Yes, it would return a 401 `invalid_request` error.